go/vt/vtgate/engine/route.go

```diff
@@ -455,7 +455,10 @@ func (route *Route) execInsertUnsharded(vcursor VCursor, queryConstruct *queryin
 	if err != nil {
 		return nil, fmt.Errorf("execInsertUnsharded: %v", err)
 	}
-	result.InsertID = uint64(insertid)
+
+	if insertid != 0 {
+		result.InsertID = uint64(insertid)
```

Comment:
I agree with @harshit-gangal on this one.

This is non intuitive and needs a comment.

My guess is: You're overwriting the `InsertID` returned by MySQL/vttablet only if the sequence returned a non zero value?

I assume this because it looks like `handleGenerate` returns an ID from the Vitess sequence? (That code lacks comments as well, btw.)